src/SignalR/clients/java/signalr/core/src/main/java/com/microsoft/signalr/HubConnection.java

```diff
@@ -349,7 +349,10 @@ public Completable start() {
                     this.state.unlock();
                 }
 
-                localStart.onError(error);
+                // this should be thread-safe as there shouldn't be any active calls to the accessTokenProvider
+                if (!(localStart.hasComplete() || localStart.hasThrowable())) {
```

Comment:
How could localStart already be completed here?

Reply:
I think I mistook it for `tokenCompletable` based on the comment 😆